### PR TITLE
Policy improvements

### PR DESCRIPTION
**Pull request by RCasatta**
#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [x] I've added tests for the new feature
* [x] I've added docs for the new feature
* [x] I've updated `CHANGELOG.md`

### CODE REVIEW

File: src/descriptor/policy.rs [hunk 1]
```diff
@@ -444,7 +452,7 @@ pub struct Policy {
     /// Type of this policy node
     #[serde(flatten)]
     pub item: SatisfiableItem,
-    /// How a much given PSBT already satisfies this polcy node **(currently unused)**
+    /// How a much given PSBT already satisfies this policy node in terms of signatures
```
Comment (notmandatory) [inline src/descriptor/policy.rs]: a little typo fix suggestion:  How much a given PSBT already ...

Comment (RCasatta) [reply src/descriptor/policy.rs]: fixed

File: src/descriptor/policy.rs [hunk 2]
```diff
@@ -444,7 +452,7 @@ pub struct Policy {
     /// Type of this policy node
     #[serde(flatten)]
     pub item: SatisfiableItem,
-    /// How a much given PSBT already satisfies this polcy node **(currently unused)**
+    /// How a much a given PSBT already satisfies this policy node in terms of signatures
```
Comment (notmandatory) [inline src/descriptor/policy.rs]: small typo here.. should read "How much a given PSBT..."

Comment (RCasatta) [reply src/descriptor/policy.rs]: lol, sorry, my brain can process one " a " at a time

**Review by afilini — APPROVED** [review]

**Pull request merged**